app/src/main/java/org/mozilla/fenix/compose/button/TextButton.kt

```diff
@@ -0,0 +1,66 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.compose.button
+
+import android.content.res.Configuration
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.sp
+import org.mozilla.fenix.theme.FirefoxTheme
+import org.mozilla.fenix.theme.Theme
+
+/**
+ * Base component for text-only buttons.
+ *
+ * @param text The button text to be displayed.
+ * @param onClick Invoked when the user clicks on the button.
+ * @param modifier [Modifier] Used to shape and position the underlying [androidx.compose.material.TextButton].
+ * @param textColor [Color] to apply to the button text.
+ */
+@Composable
+fun TextButton(
+    text: String,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+    textColor: Color = FirefoxTheme.colors.textAccent,
+) {
+    androidx.compose.material.TextButton(
+        onClick = onClick,
+        modifier = modifier,
+    ) {
+        Text(
+            text = text.uppercase(),
```

Comment:
Any reason why we dropped the `Locale.getDefault()`? It probably makes sense to consider the locale.

Reply:
My impression is `uppercase()` is using the default locale when one isn't passed-in.

Reply:
"[Uppercase] Returns a copy of this string converted to upper case using Unicode mapping rules of the invariant locale."

https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/uppercase.html

Reply:
Looking around our codebase, I would feel more comfortable passing the `Locale.getDefault()` from my search for `lowercase`, `capitalize` in both AC/Fenix.

Reply:
I'm fine with doing that to keep it consistent until we decide to re-examine it all